### ISSUE TITLE
1.18.2: ChatControl initializing placeholder hooks late.

### ISSUE THREAD

**Issue by mibby** [kangarko/ChatControl#1861]
Minecraft Version: 1.18.2
MySQL: true
BungeeCord: false

[Paper 1.18.2 dev 310](https://papermc.io/downloads) & [Purpur 1.18.2 dev 1612](https://purpurmc.org/downloads)
[ChatControlRed v10.10.5](https://www.mc-market.org/resources/18217/)
[ProtocolLib dev 553](https://ci.dmulloy2.net/job/ProtocolLib/)
[PlaceholderAPI dev 152](https://ci.extendedclip.com/job/PlaceholderAPI/)

I upgraded from ChatControlRed v10.10.2 directly to ChatControlRed v10.10.5. I noticed that during automated restarts, players who are reconnected do not have their name properly replaced in the ChatControl join message anymore. This use to work fine so I believe ChatControl is now hooking late instead of during initial server initialization.

Player connections are proxied through BungeeCord until the server is fully restarted and "done", in post-initialized, until the connection is sent back to the server for login. This issue occurs under both Paper & Purpur.

The join.rs message is configured the following;
```
group default
message: 
- login
```

The login.yml format is configured the following;
```
Parts:
  Default:
    Message: '&8[&2+&8] {player_displayname} &fhas &2joined &fthe server.'
    Hover:
    - '&f&lName:&r {player}'
```

``{player_displayname}`` variable I believe comes from PlaceholderAPI's player expansion. Hovering over the broken message outputted _does_ show the correct player name in the ChatControl hover ``{player}`` placeholder, which leads me to believe it's ChatControl not hooking with PlaceholderAPI fully. The login message begins to work after a couple seconds for normal connections or if you have 10+ people reconnecting at the same time, half of the later messages would appear fine.

![link](https://i.imgur.com/Y0n8utb.png)

Using ChatControl's native ``{display_name}`` variable seems to work around the issue. I just figured I'd notify you of this regression for anyone who does rely entirely on PAPI placeholders in join messages.

[debug.zip](https://github.com/kangarko/ChatControl-Red/files/8549547/debug.zip)

**Comment by kangarko** [comment]
Thanks for reporting. 

That's odd since I did not edit anything closely replaced to variables since that version. Anyways try setting your Debug key, in settings.yml of ChatControl Red to [placeholder-api] and it should print out the following two messages depending on the availability of PAPI at the time the function is called, for your case, during the restart. 

Let me know if something does not play right -- this will be available in the next version:

![a](https://i.imgur.com/QcLvaRH.png)

**Comment by ElBananaa** [comment]
@kangarko I believe PlaceholderAPI did change the methods used to get placeholders (which means the method you are currently using are deprecated). As (I believe) the deprecated methods will be removed in PlaceholderAPI v2.13 and were implemented somewhere in PAPI v2.10, it may be a good idea to update your implementation. I guess it can only benefit the plugin and at some point you'll be forced to update, so doing it now or in the few weeks/months wouldn't be a bad idea anyways.

**Comment by kangarko** [comment]
Thanks, really busy with recording at the moment so I will have a look once they remove the old methods.
